### DISCUSSION TITLE
Is there anyway to make AppShell side panels user-resizable?

### DISCUSSION THREAD

**Discussion by shakiba** [mantinedev/mantine#6878] [Q&A]
Hi, I wonder if there is anyway to make AppShell side panels resizable, similar to [react-split-pane](https://github.com/tomkp/react-split-pane)?

**Comment by rtivital** [comment]
No, AppShell component does not support that, use react-split-pane library to achieve that.

**Reply by shakiba** [reply]
Thanks for letting me know; is it just not a priority, or is there any reason not to implement it?

**Comment by Athou** [comment]
Hi @rtivital,

Now that Mantine has an integrated Splitter component, would you be open to reconsidering this?
I'm using react-draggable to make the [nav panel of CommaFeed resizeable](https://github.com/Athou/commafeed/blob/master/commafeed-client/src/pages/app/Layout.tsx#L280) but it would be awesome if Mantine integrated this natively.

Thanks for Mantine, it's a great library!

**Comment by saadpocalypse** [comment]
Hi @Athou,

Now that Mantine has the native Splitter component (introduced in 9.x), you can achieve resizable side panels by combining `AppShell` and `Splitter` via composition.

Instead of trying to make `AppShell.Navbar` itself draggable (which gets complex with responsive breakpoints and layouts), you can use `AppShell` to layout your header/footer, and then place a `Splitter` component inside `<AppShell.Main>` to manage the sidebar/content split.

Here is what I had in mind.

```tsx
import { AppShell, Splitter } from '@mantine/core';

export function ResizableAppLayout() {
  return (
    <AppShell header={{ height: 60 }} padding={0}>
      <AppShell.Header>
        {/* Header content here */}
      </AppShell.Header>
      
      <AppShell.Main>
        {/* Splitter handles the horizontal partition of the main area */}
        <Splitter orientation="horizontal" h="calc(100vh - 60px)">
          {/* Navigation / Sidebar Pane */}
          <Splitter.Pane defaultSize={300} min={150} max={500}>
            <div style={{ padding: 'md' }}>
              {/* Navigation content here */}
            </div>
          </Splitter.Pane>
          
          {/* Main Content Pane */}
          <Splitter.Pane>
            <div style={{ padding: 'md' }}>
              {/* Main content here */}
            </div>
          </Splitter.Pane>
        </Splitter>
      </AppShell.Main>
    </AppShell>
  );
}
```

This keeps layout concerns isolated. `AppShell` handles the top/bottom frames, and the native `Splitter` handles the resizable panels, constraints, and touch/keyboard accessibility.

Correct me if I'm wrong, I'd be willing to open a PR for this.
